src/loaders/xm_load.c

```diff
@@ -417,7 +417,7 @@ static int oggdec(struct module_data *m, HIO_HANDLE *f, struct xmp_sample *xxs,
 	n = stb_vorbis_decode_memory(data, len, &ch, &rate, &pcm16);
 	free(data);
 
-	if (n <= 0 || ch != 1) {
+	if (n < 0 || ch != 1) {
 		free(pcm16);
 		return -1;
```

Comment:
It would probably make a lot more sense to just return 0 here in the case of n==0. That would also avoid the special case for realloc.

Reply:
Seconded, I think something like
```C
if (n < 0 || ch != 1) {
  [...]
} else if (n == 0) {
  xxs->len = 0;
  free(pcm16);
  return 0;
}
```
ought to be safe and is consistent with how zero length samples are handled elsewhere. `stb_vorbis_decode_memory` *will* allocate a buffer even if the decoded length is 0.

Reply:
I have now changed the code as @AliceLR suggest.